### PR TITLE
Add proper capability parsing

### PR DESCRIPTION
**Pull request by zklapow**
As the title says.

@cimmyv @brian5021

### CODE REVIEW

File: src/main/java/com/hubspot/imap/protocol/capabilities/Capabilities.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        .collect(Collectors.toList());
+
+    return new Capabilities(capabilities);
+  }
+
+  private static Capability parseSingle(String capability) {
+    // We need to split the mechanism out for auth capabilities
+    if (capability.startsWith("AUTH")) {
+      String[] parts = capability.split("=");
+      if (parts.length < 2) {
```
Comment (brian5021) [inline src/main/java/com/hubspot/imap/protocol/capabilities/Capabilities.java]: What exactly is this logic trying to determine? For `StandardCapabilities.AUTH` we would expect capabilities to be "AUTH=" ?

File: src/main/java/com/hubspot/imap/protocol/response/tagged/CapabilityResponse.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  class Builder extends TaggedResponse.Builder implements CapabilityResponse {
+    private Capabilities capabilities;
+
+    public CapabilityResponse fromResponse(TaggedResponse input) {
+      capabilities = input.getUntagged().stream()
+          .filter(o -> o instanceof UntaggedCapabilityResponse)
+          .map(o -> ((UntaggedCapabilityResponse) o))
+          .map(UntaggedCapabilityResponse::getCapabilities)
+          .findFirst()
```
Comment (brian5021) [inline src/main/java/com/hubspot/imap/protocol/response/tagged/CapabilityResponse.java]: Do we need to sort first before using `findFirst()`?

Comment (zklapow) [reply src/main/java/com/hubspot/imap/protocol/response/tagged/CapabilityResponse.java]: there will never actually be more than one element here, this is merely a convenience

File: src/main/java/com/hubspot/imap/protocol/response/untagged/UntaggedCapabilityResponse.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  }
+
+  @Override
+  public UntaggedResponseType getType() {
+    return UntaggedResponseType.CAPABILITY;
+  }
+
+  @Override
+  public String getMessage() {
+    return null;
```
Comment (brian5021) [inline src/main/java/com/hubspot/imap/protocol/response/untagged/UntaggedCapabilityResponse.java]: Is null preferable to empty string?

File: src/main/java/com/hubspot/imap/protocol/capabilities/Capabilities.java (hunk 1 above)
Comment (zklapow) [reply src/main/java/com/hubspot/imap/protocol/capabilities/Capabilities.java]: yes, auth capabilities are specified as `AUTH=`

**Pull request merged**